src/hgvs/assemblymapper.py

```diff
@@ -241,6 +243,26 @@ def _alt_ac_for_tx_ac(self, tx_ac):
         assert len(alt_acs) == 1, "Should have exactly one alignment at this point"
         return alt_acs[0]
 
+    def _replace_reference(self, var):
```

Comment:
Yes. In order to get the reference value for these variants we need to first map them back to the `var_g` form which requires an assembly mapper afaik. We don't actually call the `*_to_g` function from the AM because that includes additional normalization. Instead we use the AM to get the `alt_ac` and then call `*_to_g` from the VM with the `alt_ac` and `alt_aln_method` from the AM. In all other cases this method simply calls  `_replace_reference` from the variant mapper instead. This is all so that we can get a portion of the reference sequence from the intronic section.

Reply:
The `AssemblyMapper` has `VariantMapper` as a parent class. In that light, do we need to duplicate this method? Also, the AM has a `normalize` property. Could that get used to prevent what seem to be an undesired side-effects? I assume the intent is not to call `_maybe_normalize`?

Reply:
Hi @andreasprlic,

I don't think I understand your first question. This `_replace_reference` method overrides the method of the same name from the parent `VariantMapper` class. It has the same purpose and type signature. The only difference is that I've added logic to handle `c` and `n` type variants with non-zero offsets.

I thought about disabling `self.normalize` temporarily in `AssemblyMapper` to avoid calling the `_maybe_normalize` method but I would also have to make sure the original value of `self.normalize` is replaced afterward, probably with a `try...finally` block. I thought it was simpler to call the underlying functions directly but I can rework this if you'd prefer.

Thanks!